### PR TITLE
bpf: gate egressgw datapath on separate defines

### PR DESCRIPTION
**Pull request by lmb**
bpf: gate egressgw datapath on separate defines

    Add a new define ENABLE_EGRESS_GATEWAY_COMMON which enables the common
    egress gateway code. The existing ENABLE_EGRESS_GATEWAY flag now controls
    whether to check for egress gateway policies with static gateway IPs. This
    makes it easier to extend the codebase to with alternative implementations.

    Signed-off-by: Lorenz Bauer <lmb@isovalent.com>

bpf: move CT checks into egress_gw_request_needs_redirect

    Move the CT reply check into egress_gw_request_needs_redirect so that
    alternative implementations can have their own CT state handling.

    Signed-off-by: Lorenz Bauer <lmb@isovalent.com>

### CODE REVIEW

**Comment by lmb** [comment]
Needs a rebase on top of #26952 since that already moves the code to a separate file.

File: bpf/bpf_lxc.c [hunk 1]
```diff
@@ -1065,7 +1065,7 @@ static __always_inline int handle_ipv4_from_lxc(struct __ctx_buff *ctx, __u32 *d
 		}
 	}
 
-#ifdef ENABLE_EGRESS_GATEWAY
+#ifdef ENABLE_EGRESS_GATEWAY_COMMON
```
Comment (jibi) [inline bpf/bpf_lxc.c:1068]: nit: let's just switch all of them to `#if defined()`?

File: bpf/lib/egress_gateway.h [hunk 2]
```diff
@@ -8,7 +8,11 @@
 
 #include "maps.h"
 
-#ifdef ENABLE_EGRESS_GATEWAY
+#if defined(ENABLE_EGRESS_GATEWAY)
+#define ENABLE_EGRESS_GATEWAY_COMMON
+#endif
+
+#ifdef ENABLE_EGRESS_GATEWAY_COMMON
```
Comment (jibi) [inline bpf/lib/egress_gateway.h:15]: ditto

**Review by jibi — APPROVED** [review]
LGTM! just a non blocking nit

**Pull request merged**